### PR TITLE
[DOC] Add a page to introduce DPCT API-Mapping query feature.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: docs/dev_guide/API-Mapping-query-guide.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+API-Mapping Query Guide
+=======================
+
+|tool_name| provides an option for users to get which SYCL API will be used when
```
Comment (zhimingwang36) [inline docs/dev_guide/API-Mapping-query-guide.rst]:
```suggestion
|tool_name| provides the functionality to query functionally compatible SYCL API for some CUDA API.  The API query functionality can assist user on manual code migration and help user understand the SYCL API.

```

File: docs/dev_guide/API-Mapping-query-guide.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
+API-Mapping Query Guide
+=======================
+
+|tool_name| provides an option for users to get which SYCL API will be used when
+migrating a CUDA API.
+
+For example:
```
Comment (zhimingwang36) [inline docs/dev_guide/API-Mapping-query-guide.rst]:
```suggestion
API mapping query example:

```

File: docs/dev_guide/API-Mapping-query-guide.rst (hunk 1 above)
Comment (tangjj11) [reply docs/dev_guide/API-Mapping-query-guide.rst]: fixed

File: docs/dev_guide/API-Mapping-query-guide.rst (hunk 2 above)
Comment (tangjj11) [reply docs/dev_guide/API-Mapping-query-guide.rst]: fixed

File: docs/dev_guide/API-Mapping-query-guide.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+   .. code-block:: bash
+
+      $ dpct --query-api-mapping=cudaMalloc
+      CUDA API:
+        cudaMalloc(pDev /*void ***/, s /*size_t*/);
+      Is migrated to:
+        *pDev = (void *)sycl::malloc_device(s, dpct::get_in_order_queue());
+
+- You can use ``tab`` button to auto-complete the CUDA API names.
```
Comment (zhimingwang36) [inline docs/dev_guide/API-Mapping-query-guide.rst]: I think this line apply for linux only.

Comment (tangjj11) [reply docs/dev_guide/API-Mapping-query-guide.rst]: fixed

**Comment by jinz2014** [comment]
"assist user on" could be "assist users on". Thanks for adding the query feature.

**Comment by tangjj11** [comment]
> "assist user on" could be "assist users on". Thanks for adding the query feature.

Fixed.

**Review by zhimingwang36 — APPROVED** [review]
LGTM (Please further ask Doc team (Kristal) to review it.

File: docs/dev_guide/API-Mapping-query-guide.rst [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        *pDev = (void *)sycl::malloc_device(s, dpct::get_in_order_queue());
+
+- You can use ``tab`` button to auto-complete the CUDA API names (In Linux OS only).
+
+    .. code-block:: bash
+
+        $ dpct --query-api-mapping=cudaMa (press tab)
+        $ dpct --query-api-mapping=cudaMalloc
+
+- |tool_name| supports fuzzy match of CUDA API names.
```
Comment (lapel22) [inline docs/dev_guide/API-Mapping-query-guide.rst]: Suggest changing "fuzzy match" to "fuzzy matching"

File: docs/dev_guide/API-Mapping-query-guide.rst [hunk 5]
```diff
@@ -0,0 +1,34 @@
+API-Mapping Query Guide
+=======================
+
+|tool_name| provides the functionality for users to query functionally
+compatible SYCL API for some CUDA API. The API query functionality can assist
+users on manual code migration and help users understand the SYCL API.
```
Comment (lapel22) [inline docs/dev_guide/API-Mapping-query-guide.rst]: Consider changing to "manual code migration, as well as help users understand the SYCL API" for flow

File: docs/dev_guide/API-Mapping-query-guide.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+   .. code-block:: bash
+
+      $ dpct --query-api-mapping=cudaMalloc
+      CUDA API:
+        cudaMalloc(pDev /*void ***/, s /*size_t*/);
+      Is migrated to:
+        *pDev = (void *)sycl::malloc_device(s, dpct::get_in_order_queue());
+
+- You can use ``tab`` button to auto-complete the CUDA API names (In Linux OS only).
```
Comment (lapel22) [inline docs/dev_guide/API-Mapping-query-guide.rst]: Change to "On a Linux OS, you can use the ``tab`` button" to auto-complete the CUDA API names."

**Review by lapel22 — CHANGES_REQUESTED** [review]
Suggested some small changes for grammar/flow, but other than those small changes, this looks good!

File: docs/dev_guide/API-Mapping-query-guide.rst (hunk 4 above)
Comment (tangjj11) [reply docs/dev_guide/API-Mapping-query-guide.rst]: fixed

File: docs/dev_guide/API-Mapping-query-guide.rst (hunk 5 above)
Comment (tangjj11) [reply docs/dev_guide/API-Mapping-query-guide.rst]: fixed

File: docs/dev_guide/API-Mapping-query-guide.rst (hunk 6 above)
Comment (tangjj11) [reply docs/dev_guide/API-Mapping-query-guide.rst]: fixed

**Pull request merged**